.github/workflows/check-spelling.yml

```diff
@@ -0,0 +1,17 @@
+name: Check spelling # cSpell:ignore refcache
+
+on:
+  pull_request:
+
+jobs:
+  check-formatting:
+    name: Check spelling
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: streetsidesoftware/cspell-action@v2
+        with:
+          files: |
+            content/**/*
+          config:
+            .vscode/cspell.json
```

Comment:
this one is much faster than doing it via `npm run check...`, it also only checks files that have been changed via the PR. And it provides inline feedback:

<img width="1354" alt="Screenshot 2023-07-05 at 10 53 06" src="https://github.com/open-telemetry/opentelemetry.io/assets/1519757/d8b6709b-25e5-4cbb-89dd-2461a2b20244">